### ISSUE TITLE
The provider data are not refreshed when the user log again (so no new accesstoken)

### ISSUE THREAD

**Issue by junit38** [meanjs/mean#511]

**Comment by ilanbiala** [comment]
Please highlight your code using GFM.

**Comment by codydaig** [comment]
@junit38 Are you still having issues?

**Comment by junit38** [comment]
Yes, the user provider's data are no refreshed on each signin.
The lines to change is now in the file: 
modules/users/server/controllers/users/users.authentication.server.controller.js

at line 175-177:

```
} else {
          return done(err, user);
}
```

**Comment by ilanbiala** [comment]
@lirantal @junit38 any idea whether this is still an issue?

**Comment by lirantal** [comment]
I didn't eventually get to review that one.
If anyone has any social login configured we can easily test by changing some profile detail after being logged-in once and then logout/login to see if that new value is updated on meanjs.

**Comment by junit38** [comment]
Yes there is still and issue on the last version of the framework.
Just sign in using facebook strategy and then logout/login to see that the data did'nt not change.
As @lirantal suggested.

**Comment by mleanos** [comment]
@junit38 Can you explain what you're experiencing a little more clearly?

How can I reproduce? What data is not getting updated?

This is how I've tried to reproduce, and couldn't:

1) Created a new local provider account
2) Added Facebook as a provider
3) Logged Out
4) Logged back in using the local provider user/pass method
5) Facebook provider was still linked to the account

OR

1) Created a new local provider account
2) Added Facebook as a provider
3) Updated Last Name
4) Logged Out
5) Logged in again, and Last Name reflected my change 

**Repeated above step, but logging in again as Facebook provider; same result**

OR

1) Created a new account using Facebook provider
2) Updated Last Name
3) Logged out
4) Logged back in, and Last Name reflected my update

**Comment by junit38** [comment]
1) Create a Facebook account using your Facebook signup.
2) See the accessToken
3) Logout
3) Login again

The accesToken is not refreshed. (You can try to change your Facebook name to see but this is a trap because you can change it only every 60 days....)

This is all the ProviderData which are not refresh.

So you can't use the Facebook accessToken to do action on the user Facebook account by example (the access token expiration limit is above 2months but it's still an issue)

**Comment by mleanos** [comment]
@junit38 Ok, so if you change some data on your social media (Facebook in this case) account, then your meanjs account isn't updated the next tine you log in with that provider?

I'll try to confirm this. But I suspect it is an issue because I don't see anywhere in the code, that updates the provider data; just as you originally pointed out. 

Can you submit a PR for the fix?

**Comment by mleanos** [comment]
@junit38 Well, other than my profile picture, I didn't see anything in my Facebook profile that I could that would reflect in my meanjs account; I didn't want to change my Profile name.

I did verify that when I log in again using the Facebook provider, the meanjs account isn't updated; I think the line that you wanted to change is the correct place to add the update. 

Do you want to handle that with a PR?

**Comment by jloveland** [comment]
I also verified that when I change the my profile image in facebook, the photo url changes in `profileImageURL`, but not the `providerData.picture.url`. So, this leads me to believe that if the user accessToken from facebook changes (i.e. after 2 months) then the token won't get updated in the `providerData.accessToken`field. 

@junit38 are you able to submit a PR to update the providerData after authentication if it's changed?

**Comment by mleanos** [comment]
None of the data is updated from the provider data when the user is authenticated again.

@jloveland Brings up a good point about the Token not getting updated. I think we should have this updated at each authentication. However, I don't think any other data should be pulled over from the provider data. 

I think it makes more sense to have a "Sync" button, if the user wants to pull over data from their provider. Otherwise, I think treating their meanjs data as the *master* should be the way it's done.

**Comment by junit38** [comment]
Yes i'm on it.

**Comment by ilanbiala** [comment]
@junit38 can you submit a PR for this against the current master?

**Comment by junit38** [comment]
Yes but it appears that I don't have the rights.

**Comment by ilanbiala** [comment]
@junit38 fork and make changes and then click compare and create a pull request.
